File: CODIGO_C++/ACT3_1.cpp
#include <iostream>
#include <string>

using namespace std;

//Definimos la clase "Animal"
class Animal {
private:
    string nombre;
    int edad;
    string tipo;

public:
    //Constructor de la clase Animal que inicializa con los atributos
    Animal(string n, int e, string t) : nombre(n), edad(e), tipo(t) {}

    //Metodo para alimentar al animal
    void alimentar() {
        cout<<nombre<<" esta siendo alimentado."<<endl;
    }

    //Metodo para que el animal haga un sonido
    void hacer_sonido() {
        cout<<nombre<<" hace un sonido caracteristico."<<endl;
    }

    //Metodo para mostrar la edad del animal
    void mostrar_edad() {
        cout<<nombre<<" tiene "<<edad<<" years."<<endl;
    }

    //Metodo para mostrar el tipo de animal
    void mostrar_tipo() {
        cout<<nombre<<" es un "<<tipo<<"."<<endl;
    }
};

int main() {
    //Creacion de objetos de la clase Animal
    Animal perro("Puchito", 7, "Perro");
    Animal gato("Dieguito", 4, "Gato");
    Animal pajaro("Piquito", 2, "Pajaro");

    //Interaccion con el perro
    perro.alimentar();
    perro.hacer_sonido();
    perro.mostrar_edad();
    perro.mostrar_tipo();

    //Interaccion con el gato
    gato.alimentar();
    gato.hacer_sonido();
    gato.mostrar_edad();
    gato.mostrar_tipo();

    //Interaccion con el pajaro
    pajaro.alimentar();
    pajaro.hacer_sonido();
    pajaro.mostrar_edad();
    pajaro.mostrar_tipo();

    return 0;
}

File: CODIGO_PYTHON/ACT3_1.py
class Animal:
    def __init__(self, nombre, edad, tipo):
        self.nombre = nombre
        self.edad = edad
        self.tipo = tipo

    def alimentar(self):
        print(self.nombre, "está siendo alimentado.")

    def hacer_sonido(self):
        print(self.nombre, "hace un sonido característico.")

    def mostrar_edad(self):
        print(self.nombre, "tiene", self.edad, "años.")

    def mostrar_tipo(self):
        print(self.nombre, "es un", self.tipo + ".")


if __name__ == "__main__":
    #Creacion de objetos de la clase Animal
    perro = Animal("Puchito", 7, "Perro")
    gato = Animal("Dieguito", 4, "Gato")
    pajaro = Animal("Piquito", 2, "Pájaro")

    #Interaccion con el perro
    perro.alimentar()
    perro.hacer_sonido()
    perro.mostrar_edad()
    perro.mostrar_tipo()

    #Interaccion con el gato
    gato.alimentar()
    gato.hacer_sonido()
    gato.mostrar_edad()
    gato.mostrar_tipo()

    #Interaccion con el pájaro
    pajaro.alimentar()
    pajaro.hacer_sonido()
    pajaro.mostrar_edad()
    pajaro.mostrar_tipo()
